test/common_nn.py

```diff
@@ -854,6 +856,22 @@ def test_cuda(self, test_case):
                 for cpu_d_p, gpu_d_p in zip(cpu_param[1], gpu_param[1]):
                     test_case.assertEqual(cpu_d_p, gpu_d_p, 2e-4)
 
+            # Run double-backwards on CPU and GPU and compare results
+            # TODO: BatchNorm double-backwards is currently broken, see
+            # #4422; remove this when it is fixed.
+            if self.check_gradgrad and not self.get_name().startswith('test_BatchNorm'):
+                for i in range(5):
+                    cpu_output_t = cpu_output.data if isinstance(cpu_output, Variable) else cpu_output
+                    cpu_gradOutput = Variable(cpu_output_t.clone().bernoulli_(), requires_grad=True)
```

Comment:
Why bernoulli?

Reply:
Ditto as above.

Reply:
Bernoulli doesn't seem like a good choice of random function: 0s ignore inputs and 1s can mask scaling bugs. Something like `randn` might be better.

Reply:
I assume this reasoning applies to the backwards calculation too. I thought there was some clever reason bernoulli was picked but maybe not...

Reply:
is there a reason we are (mostly) duplicating this sort of code?

Reply:
Mostly because I was too lazy to think of a good abstraction in this case. I suppose I can write a general purpose harness for comparing two functions/modules for equivalent behavior, and another for GPU'ifying a module/input pair... if you really want me to...

Reply:
That should actually be pretty easy - do the `gradgradcheck` trick. Write a function that takes another function and compares outputs and grads, then pass `torch.autograd.grad(module(x), x)` as the callable to test double backward.

Reply:
The gradgradcheck trick is a bit difficult to do because it assumes that we are interacting with the module under test functionally, and the existing tests are not doing this.

My new plan is to delete the backward comparison code, and just run gradcheck/gradgradcheck on the CUDA kernels.

Reply:
In offline conversation @apaszke and I came to the conclusion that it is more trouble than it is worth trying to deduplicate this code, because we have to use the stateful module API (because this code is exercised by legacy NN which doesn't support functional gradient computation), which means you can't "just" do the gradgradcheck trick, you need a version of this trick which wraps a module not a function.

Reply:
Update: we have a different strategy, it'll be in the new patch